src/app/services/app-auth.service.ts

```diff
@@ -0,0 +1,70 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { Router } from '@angular/router';
+import { BehaviorSubject } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+export type User = {
+  id: string;
+  name: string;
+  balance: number;
+};
+
+/**
+ * This is not the actual implementation of authentication service,
+ * this is only example, be careful with reading all of the notes inside.
+ */
+@Injectable({
+  providedIn: 'root',
+})
+export class AppAuthService {
+  private isAuthenticated$$ = new BehaviorSubject<boolean>(false);
+
+  private currentUser$$ = new BehaviorSubject<User | null>(null);
+  public currentUser$ = this.currentUser$$.asObservable();
+
+  /**
+   * In the production version of auth service it should be like
+   * public isAuthenticated$ = this.authServiceFromFoundation.isAuthenticated;
+   */
+  public isAuthenticated$ = this.isAuthenticated$$.asObservable();
+
+  /**
+   * In the production version of app auth service there will be only one dependency
+   * for authService from foundation-ang
+   */
+  constructor(private http: HttpClient, private router: Router) {}
+
+  /**
+   * In the production version only `redirectUri` is needed to be provided as an argument for
+   * this.authServiceFromFoundation.login({ redirectUri }) method call
+   */
+  login({ login, redirectUri }: { login: string; redirectUri: string }): void {
+    this.http
+      .post('/api/auth/login', { login, redirectUri })
```

Comment:
Thank you @yenobi for this example. It was long pending.

One suggestion here perhaps - the auth service is only going to be used for hybrid strategy i.e delegating auth responsibility to Identity(or any other IDP). But in the example here you the Angular app is directly interacting with an authentication endpoint(giving it a feeling of bas auth strategy).

Is it worth mentioning here explicitly in the comment that in production this login mechanism will be handled by customer's IDP?

Reply:
I have updated the comment here https://github.com/Backbase/golden-sample-app/pull/11/commits/edcff60a036679fe2324b632d51ebb656d0eca33

Do you think it will be enough or should I have a bigger explanation?

Reply:
I think it is enough - thanks